v1/daemon/write_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/name"
+	"github.com/google/go-containerregistry/v1/tarball"
+)
+
+func TestWriteImage(t *testing.T) {
```

Comment:
This doesn't override `getImageLoader`, so it has a dependency on docker, which I don't have installed, so I can't run the tests :)

```
write_test.go:36: Error writing image tar: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

(you probably have some `test_image_2:latest`s in your daemon 😂 )

Reply:
this is testing loading an image into the local daemon, so if you don't have docker installed this will definitely fail :)

the test is just doing a `tarball.ImageFromPath` on one of the tars checked into testdata and then retagging it as `test_image_2:latest`.

Reply:
I won't fight this too hard, but it would be nice if I could `bazel test ...` this without installing docker :) can you mock out the docker interaction for this test and add an integration test that exercises the actual daemon?

I'm not a huge fan of implicit dependencies, but if nobody else cares, it's nbd.

Reply:
no that's a good point, if we were relying on docker to always be present then what would be the point of this library :) fixed to mock out so we don't need docker to test

Reply:
Thanks!